src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -4798,13 +4798,16 @@ export class ResultsViewPageStore {
                     getOncoKbMutationAnnotationForOncoprint(mutation);
 
                 const isHotspotDriver =
+                    this.driverAnnotationSettings.hotspots &&
                     !(this.isHotspotForOncoprint.result instanceof Error) &&
                     this.isHotspotForOncoprint.result!(mutation);
                 const cbioportalCountExceeded =
+                    this.driverAnnotationSettings.cbioportalCount &&
                     this.getCBioportalCount.isComplete &&
                     this.getCBioportalCount.result!(mutation) >=
                         this.driverAnnotationSettings.cbioportalCountThreshold;
                 const cosmicCountExceeded =
+                    this.driverAnnotationSettings.cosmicCount &&
```

Comment:
so this is mostly to get the speed up when driver annotations by cosmic are disabled right (i guess that's the default)

Reply:
right. we shouldnt be using this if cosmic is disabled (this is also, I think, the cause of the issue where the loading goes backwards - its because were using a promise in `invoke` that wasn't in `await`